File: ruby/testcode/weakcipher_035.rb
require_relative 'shared'
require 'openssl'

# vuln-code-snippet start ruby_weakcipher_3des_cbc_short_key
def encrypt_3des_short(req)
  plaintext = req.body_str
  cipher = OpenSSL::Cipher.new('DES-EDE-CBC') # vuln-code-snippet vuln-line ruby_weakcipher_3des_cbc_short_key
  cipher.encrypt
  cipher.key = cipher.random_key
  cipher.iv = cipher.random_iv
  ciphertext = cipher.update(plaintext) + cipher.final
  BenchmarkResponse.ok(ciphertext.unpack1('H*'))
end
# vuln-code-snippet end ruby_weakcipher_3des_cbc_short_key

File: php/testcode/weakcipher_035.php
<?php
require_once __DIR__ . '/shared.php';

define('DES_KEY', 'weakdes!');

// vuln-code-snippet start php_weakcipher_des_global_key
function weakcipher035(BenchmarkRequest $req): BenchmarkResponse {
    $enc = openssl_encrypt($req->param('data'), 'DES-CBC', DES_KEY, 0, random_bytes(8)); // vuln-code-snippet vuln-line php_weakcipher_des_global_key
    return BenchmarkResponse::ok($enc);
}
// vuln-code-snippet end php_weakcipher_des_global_key
